patches/components-history-core-browser-sync-typed_url_sync_bridge.cc.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/components/history/core/browser/sync/typed_url_sync_bridge.cc b/components/history/core/browser/sync/typed_url_sync_bridge.cc
+index eb0494be3942e1ae0eab89ca56d64be60b493ab5..15de8baa4f3147450566d287656d5b8baf924742 100644
+--- a/components/history/core/browser/sync/typed_url_sync_bridge.cc
++++ b/components/history/core/browser/sync/typed_url_sync_bridge.cc
+@@ -68,6 +68,7 @@ std::string GetStorageKeyFromURLRow(const URLRow& row) {
+ }
+ 
+ bool HasTypedUrl(const std::vector<VisitRow>& visits) {
++  BRAVE_HAS_TYPED_URL_RETURN_TRUE_IF_SEND_ALL_HISTORY_ENABLED
+   return base::ranges::any_of(visits, [](const VisitRow& visit) {
+     return ui::PageTransitionCoreTypeIs(visit.transition,
+                                         ui::PAGE_TRANSITION_TYPED);
+@@ -527,6 +528,7 @@ bool TypedURLSyncBridge::WriteToTypedUrlSpecifics(
+     // Walk the passed-in visit vector and count the # of typed visits.
+     for (const VisitRow& visit : visits) {
+       // We ignore reload visits.
++      BRAVE_WRITE_TO_TYPED_URL_SPECIFICS_DONT_IGNORE_RELOADS
+       if (PageTransitionCoreTypeIs(visit.transition,
```

Comment:
ptal? looks like we can fully remove this patch and make things a little clearer:
https://gist.github.com/goodov/2becb1cd4b5930af436d1c6592f32084

Reply:
This is much better than what I have now. Updating the PR, thanks.

Reply:
Yeah, this works incredible. Updated at 
https://github.com/brave/brave-core/pull/16705/commits/875844cd3a0e02a8c06a780660fd45e1dd4bae3a

But after DM with @darkdh  we found that we need to have some restriction for non-TYPED transitions, otherwise the server will be overloaded and browser's main work will be just sending history to sync.

So I am  to applying this logic 
```
bool BraveTypedURLSyncBridge::ShouldSyncVisit(const URLRow& url_row,
                              ui::PageTransition transition) {
  if (base::FeatureList::IsEnabled(
          brave_sync::features::kBraveSyncSendAllHistory)) {
    return url_row.visit_count < 20 ||
           (url_row.visit_count % 10) == 0;
  }
  return TypedURLSyncBridge::ShouldSyncVisit(url_row.typed_count(), transition);
}
```
and maybe I bring back one patch

Reply:
I had to add a new patch here https://github.com/brave/brave-core/pull/16705/commits/ed7ae4b90a88ea3f3b66cbc71880bf87f7ed2161